scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -401,44 +406,56 @@ private void receiveNextDatagramFromNetwork() throws IOException {
 		for (Record record : records) {
 			try {
 				LOGGER.log(Level.FINEST, "Received DTLS record of type [{0}]", record.getType());
-				
+
 				switch(record.getType()) {
 				case APPLICATION_DATA:
-					processApplicationDataRecord(peerAddress, record);
+					processApplicationDataRecord(record);
 					break;
 				case ALERT:
-					processAlertRecord(peerAddress, record);
+					processAlertRecord(record);
 					break;
 				case CHANGE_CIPHER_SPEC:
-					processChangeCipherSpecRecord(peerAddress, record);
+					processChangeCipherSpecRecord(record);
 					break;
 				case HANDSHAKE:
-					processHandshakeRecord(peerAddress, record);
+					processHandshakeRecord(record);
+					break;
 				}
-			} catch (InvalidMacException e) {
-				// this means that the message from the record could not be authenticated
-				// maybe because the record has been sent in a forged UDP datagram by an attacker
-				// the DTLS 1.2 spec section 4.1.2.7 (see http://tools.ietf.org/html/rfc6347#section-4.1.2.7)
-				// advises to silently discard such records
-				LOGGER.log(Level.FINE, "Discarding [{0}] record from peer [{1}]: MAC validation failed",
-						new Object[]{record.getType(), peerAddress});
-			} catch (GeneralSecurityException e) {
-				// this means that the message could not be decrypted, e.g. because the JVM does
-				// not support the negotiated cipher algorithm or the ciphertext has the wrong block size etc.
-				// the DTLS 1.2 spec section 4.1.2.7 (see http://tools.ietf.org/html/rfc6347#section-4.1.2.7)
-				// advises to silently discard such records
-				LOGGER.log(Level.FINE, "Discarding [{0}] record from peer [{1}]: {2}",
-						new Object[]{record.getType(), peerAddress, e.getMessage()});
-			} catch (HandshakeException e) {
-				if (AlertLevel.FATAL.equals(e.getAlert().getLevel())) {
-					LOGGER.log(Level.INFO, "Aborting handshake with peer [{1}]: {2}",
-							new Object[]{record.getType(), peerAddress, e.getMessage()});
-					terminateOngoingHandshake(peerAddress, e.getAlert());
+			} catch (DtlsHandshakeException e) {
+				if (AlertLevel.FATAL.equals(e.getLevel())) {
+					if (LOGGER.isLoggable(Level.INFO)) {
+						StringBuilder msg = new StringBuilder("Aborting handshake with peer [")
+								.append(record.getPeerAddress()).append("]: ");
+						if (LOGGER.isLoggable(Level.FINEST)) {
+							msg.append(System.lineSeparator()).append(e);
```

Comment:
this will not display the stacktrace.
anyway I wonder what is the purpose of so much logger code :)